Makefile

```diff
@@ -246,7 +246,8 @@ build/data/%.o: data/%.s
 	$(AS) $(ASFLAGS) $< -o $@
 
 build/assets/text/%.enc.h: assets/text/%.h assets/text/charmap.txt
-	python3 tools/msgenc.py assets/text/charmap.txt $< $@
+	$(CPP) -P -dD -fpreprocessed $< > $(@:.enc.h=.h)
```

Comment:
`-dD` keeps `#define`s, which `-fpreprocessed` doesn't seem to do on it's own.

Reply:
keeps #define lines or keeps macro usage like e.g. `ITEM_ICON("\x2D")` ?
I'm not seeing a difference with or without -dD

Reply:
I mean a line line `#define wow` in the file itself. You don't see a difference since an autogenerated file won't have any defines, you'd have to add one to see it.

Reply:
I see, indeed
```
$ cpp -fpreprocessed assets/text/message_data.h > test
$ cpp -dD -fpreprocessed assets/text/message_data.h > test2
$ diff test test2
2c2
< 
---
> #define wow
```
well that's not exactly intuitive behavior, congrats on figuring that out

Reply:
For what is worth, `-fpreprocessed` breaks the build on macOS.

```
cpp -P -dD -fpreprocessed assets/text/message_data.h > build/assets/text/message_data.h
clang: error: unknown argument: '-fpreprocessed'
```

On macOS cpp is an alias for clang, and `-fpreprocessed` doesn't seem to be implemented in clang.

I fiddled a little to find a workaround, but didn't come up with anything yet.

Reply:
Feel free to drop in the discord to discuss this (link in the readme). Not many people use macOS but some may be able to help

Maybe you could try this,
replace
```mk
	$(CPP) -P -dD -fpreprocessed $< > $(@:.enc.h=.h)
```
with just
```mk
	cp $< $(@:.enc.h=.h)
```
which essentially reverts this PR's changes to the makefile. I think it may work then, maybe

Reply:
Thanks for the answer! I fixed my build by reverting the Makefile to a previous version, yes. I'll drop on Discord to see if we can find a clang-compatible setting for doing the same task.

Reply:
See #1151